src/components/cc-email-list/cc-email-list.js

```diff
@@ -0,0 +1,404 @@
+  }
+
+  _onAdd () {
+    const address = this._addressInputRef.value.value;
+
+    this.addEmailForm = {
+      state: 'idle',
+      address: {
+        value: this.addEmailForm.address.value,
+        error: validateEmailAddress(address),
+      },
+    };
+
+    if (this.addEmailForm.address.error == null) {
+      dispatchCustomEvent(this, 'add', address);
+    }
+  }
+
+  updated (_changedProperties) {
+    if (this.addEmailForm.address.error != null) {
+      setTimeout(() => this._focusAddressInput(), 0);
+    }
+  }
+
+  render () {
+    return html`
+      <cc-block>
+        <div slot="title">${i18n('cc-email-list.title')}</div>
+
+        ${when(this.emails.state === 'error', () => html`
+          <cc-error>${i18n('cc-email-list.loading.error')}</cc-error>
+        `, () => html`
+          ${this._renderPrimarySection()}
+          ${this._renderSecondarySection()}
+        `)}
+      </cc-block>
+    `;
+  }
+
+  _renderPrimarySection () {
+    const skeleton = this._isLoading();
+    const address = skeleton ? fakeString(35) : this.emails.value.primaryAddress.address;
+    const verified = skeleton ? false : this.emails.value.primaryAddress.verified;
+    const shouldDisplayResendConfirmationEmail = !skeleton && !verified;
+
+    const badgeIntent = verified ? 'success' : 'danger';
+    const badgeIcon = verified ? verifiedSvg : unverifiedSvg;
+
+    return html`
+      <cc-block-section>
+        <div slot="title">${i18n('cc-email-list.primary.title')}</div>
+        <div slot="info">${i18n('cc-email-list.primary.description')}</div>
+
+        <div class="address-line primary">
+          <div class="address">
+            <img src="${mailStarSvg}" alt=""/>
+            <span class="${classMap({ skeleton })}">${address}</span>
+          </div>
+          <cc-badge
+              intent="${badgeIntent}"
+              weight="outlined"
+              ?skeleton="${skeleton}"
+              icon-src="${badgeIcon}"
+              icon-alt=""
+          >${this._getVerifiedTagLabel()}
+          </cc-badge>
+        </div>
+        ${shouldDisplayResendConfirmationEmail ? html`
+          <cc-button
+              @cc-button:click=${this._onSendConfirmationEmail}
+              ?waiting=${this.emails.value.primaryAddress.state === 'sending-confirmation-email'}
+              link
+          >
+            ${i18n('cc-email-list.primary.action.resend-confirmation-email')}
+          </cc-button>
+        ` : ''}
+      </cc-block-section>
+    `;
+  }
+
+  _renderSecondarySection () {
+    const addresses = this._getSecondaryAddresses();
+    const markingAsPrimary = addresses.some((item) => item.state === 'marking-as-primary');
+
+    return html`
+      <cc-block-section>
+        <div slot="title">${i18n('cc-email-list.secondary.title')}</div>
+        <div slot="info">${i18n('cc-email-list.secondary.description')}</div>
+
+        <ul class="secondary-addresses">
+          ${addresses.map((secondaryAddress) => {
```

Comment:
true, We don't really need it here

Reply:
I would say: If it's a big list with many refresh => use `repeat`. Otherwise, you can stick on `map`.
This being said, when using react you just don't have a choice: you must use a repeat-like function to display lists.